app/grandchallenge/products/templates/products/partials/product_image_carousel.html

```diff
@@ -1,31 +1,33 @@
 {% if product.images.all %}
-  <div id="product_img_carousel" class="carousel slide" data-ride="carousel" data-interval="false">
-    <ol class="carousel-indicators">
-      {% for image in product.images.all %}
-        <li data-target="#product_img_carousel" data-slide-to={{ forloop.counter0 }} {% if forloop.counter0 == 0%}class="active"{% endif %}></li>
-      {% endfor %}
-    </ol>
-    <div class="carousel-inner">
-      {% for image in product.images.all %}
-        <div class="carousel-item {% if forloop.counter0 == 0 %}active{% endif %}">
-          <img class="d-block w-100" src={{ image.img.url }}>
-        </div>
-      {% endfor %}
+  {% if product.images.all.count == 1 %}
```

Comment:
`.all` is not needed here, you can just use:
```suggestion
  {% if product.images.count == 1 %}
```
By calling `.all` you first get all the objects from the database and then perform a count on them. By calling `.count` directly you create a query to the extend of `SELECT COUNT(*) from ...` which is a lot more efficient, especially on larger datasets.